File: test/services/documents/pdf/PDFMappingTest.java
package services.documents.pdf;

import com.google.common.collect.Lists;
import org.junit.Ignore;
import org.junit.Test;

import java.util.Map;

import static org.junit.Assert.*;

public class PDFMappingTest {

    @Test
    public void testMapStringValuesWithSubstringAndConcat() throws Exception {
        Map<String, String> stringStringMap = PDFMapping.mapStringValues(
                Lists.newArrayList(
                        new PDFField("field1", "a"),
                        new PDFField("field2", "b"),
                        new PDFField("field3", "c"),
                        new PDFField("field4", "abc")
                ),
                Lists.newArrayList(
                        new PDFFieldLocator("pdfField", "field1", 0, null, null, null, false),
                        new PDFFieldLocator("pdfField", "field2", 1, null, null, null, false),
                        new PDFFieldLocator("pdfField", "field3", 2, null, null, null, false),
                        new PDFFieldLocator("pdfSubSField", "field4", 0, null, 0, 1, false)
                ));

        assertEquals("a b c", stringStringMap.get("pdfField"));
        assertEquals("a", stringStringMap.get("pdfSubSField"));
    }

    @Test
    public void testReadLocators() throws Exception {
        PDFMapping.getSpec(PDFMapping.class.getClassLoader().getResourceAsStream("forms/VBA-21-0966-ARE.locators.json"));
    }
}

File: test/services/documents/pdf/PDFMappingTest.scala
package services.documents.pdf

import play.api.libs.json.JsString
import play.api.test.PlaySpecification

class PDFMappingTest extends PlaySpecification {

  "mapStringValues" should {
    "substring and concat correctly" in {
      val stringStringMap = PDFMapping.mapStringValues(
        Map(
          ("field1", JsString("a")),
          ("field2", JsString("b")),
          ("field3", JsString("c")),
          ("field4", JsString("abc"))
        ),
        Seq(
          PDFFieldLocator("pdfField", "field1", 0, None, None, None, false),
          PDFFieldLocator("pdfField", "field2", 1, None, None, None, false),
          PDFFieldLocator("pdfField", "field3", 2, None, None, None, false),
          PDFFieldLocator("pdfSubSField", "field4", 0, None, Some(0), Some(1), false)
        )
      )

      "a b c" must be equalTo stringStringMap("pdfField")
      "a" must be equalTo stringStringMap("pdfSubSField")
    }
  }
}
